src/jit/codegencommon.cpp

```diff
@@ -3981,10 +4001,12 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         {
             // A struct might be passed  partially in XMM register for System V calls.
             // So a single arg might use both register files.
+#ifndef ARM_SOFTFP
             if (isFloatRegType(regType) != doingFloat)
```

Comment:
"The incoming FP args will be in integer registers, and will probably get homed to some stack location." - so here we have LclVarDsc * varDsc; which might be a float value, right? Next the type of that value is used to form the way the value is copied and it never expected an integer argument register being copied to a floating value - either stored or just copied to a register.

How we properly avoid such a situation?